packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -0,0 +1,667 @@
+            event.preventDefault();
+            redirectTo(props.linkTo);
+          } else if (props.onClick) {
+            event.persist();
+            props.onClick<typeof gtmTracking.track>(event, gtmTracking.track);
+          }
+        }}
+      >
+        {props.children}
+      </NavLink>
+    );
+  }
+  return <>{props.children}</>;
+};
+MenuItemLink.displayName = 'MenuItemLink';
+MenuItemLink.defaultProps = menuItemLinkDefaultProps;
+
+// This component just render two divs with borders in order to differentiate
+// the settings plugin from the other ones
+const MenuItemDivider = () => (
+  <div className={styles['divider-first-item']}>
+    <div className={styles['divider-second-item']} />
+  </div>
+);
+MenuItemDivider.displayName = 'MenuItemDivider';
+
+// This component wraps `<RestrictedByPermissions>`
+// and the `<ToggleFeature>` components. However, it's necessary to have it as
+// the `<ToggleFeature>` wrapper should be rendered only if the `featureToggle`
+// prop is defined. This is because `<ToggleFeature>` will not render any
+// children if the flag is missing/not found.
+const hasMenuEveryMenuVisibilitySetToBeHidden = (
+  menuVisibilities?: TApplicationContext<{}>['menuVisibilities'],
+  namesOfMenuVisibilities?: string[]
+) =>
+  Array.isArray(namesOfMenuVisibilities) &&
+  namesOfMenuVisibilities.length > 0 &&
+  namesOfMenuVisibilities.every(
+    nameOfMenuVisibility =>
+      menuVisibilities && menuVisibilities[nameOfMenuVisibility] === true
+  );
+const isMenuItemDisabledForEnvironment = (
+  keyOfMenuItem: string,
+  disabledMenuItems?: string[]
+) => disabledMenuItems && disabledMenuItems.includes(keyOfMenuItem);
+
+type RestrictedMenuItemProps = {
+  featureToggle?: string;
+  namesOfMenuVisibilities?: string[];
+  menuVisibilities?: TApplicationContext<{}>['menuVisibilities'];
+  disabledMenuItems?: string[];
+  keyOfMenuItem: string;
+  permissions: string[];
+  actionRights?: TActionRight[];
+  dataFences?: TDataFence[];
+  children: React.ReactNode;
+};
+const restrictedMenuItemDefaultProps: Pick<
+  RestrictedMenuItemProps,
+  'permissions'
+> = {
+  permissions: [],
+};
+const RestrictedMenuItem = (props: RestrictedMenuItemProps) => {
+  // NOTE: Custom application are activated/deactivated while their
+  // visibility is not controlled via a visibiility overwrite.
+  if (
+    hasMenuEveryMenuVisibilitySetToBeHidden(
+      props.menuVisibilities,
+      props.namesOfMenuVisibilities
+    ) ||
+    isMenuItemDisabledForEnvironment(
+      props.keyOfMenuItem,
+      props.disabledMenuItems
+    )
+  )
+    return null;
+
+  const permissionsWrapper =
+    (Array.isArray(props.permissions) && props.permissions.length > 0) ||
+    (Array.isArray(props.dataFences) && props.dataFences.length > 0) ? (
+      <RestrictedByPermissions
+        permissions={props.permissions}
+        actionRights={props.actionRights}
+        dataFences={props.dataFences}
```

Comment:
The `dataFence` value provided in the menu config contains the `type` field which is typed as a `string`

<img width="252" alt="image" src="https://user-images.githubusercontent.com/1110551/73127265-224c8e00-3fbe-11ea-9dbb-89c08d383a87.png">

This is fine, however in the permissions package we used to expect a specific value like `store`. Now in the permissions package we don't have this hard constraint anymore, the `type` field is just a string.

Reply:
Got it. Disregard previous comment.